### ISSUE TITLE
iOS 10 support

### ISSUE THREAD

**Issue by romainlouvet** [appium/appium#6597]
## The problem

Appium can't launch ipa with iOS 10 beta / Xcode 8 beta.

[MJSONWP] Encountered internal error running command: Error: Could not find Automation.tracetemplate in any of the followinglocations /Applications/Xcode-beta.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate,/Applications/Xcode-beta.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.bundle/Contents/Resources/Automation.tracetemplate
    at getAutomationTraceTemplatePathWithoutRetry$ (lib/xcode.js:172:9)

## Environment

* Appium version : 1.5.3
* Desktop OS/version used to run Appium: OS X 10.11.5
* Mobile platform/version under test: iPhone 5SE with iOS 10
* Real device or emulator/simulator: Real device
Xcode 8 beta

## Link to Appium logs

https://gist.github.com/romainlouvet/098e33195ea8be2989bb0fc82e270af2

**Comment by imurchie** [comment]
Did this ipa install with Xcode 7? The installation doesn't seem to be a problem locally.

On the other hand, Xcode 8 beta does not include any automation trace template for Instruments, so it won't work with Appium at the moment. We are looking into how to get around this at the moment (any help would be appreciated).

**Comment by romainlouvet** [comment]
it's not possible to install ipa with xcode 7 / ios 10

**Comment by imurchie** [comment]
What are the logs for that situation?

**Comment by romainlouvet** [comment]
Xcode 7.3.1 / Build version 7D1014
iPhone 5SE iOS 10 Beta

https://gist.github.com/romainlouvet/6ed9e6b0e7b6e3c712e1116ccbef9e12

**Comment by romainlouvet** [comment]
If I try to manipulate my device(iOS 10) with idevicediagnostics

idevicediagnostics restart
-> ERROR: Could not connect to lockdownd, error code -5

**Comment by meirtsvi** [comment]
Compile libimobiledevice from the latest code and you're all set up:
https://github.com/libimobiledevice/libimobiledevice/issues/327

**Comment by imurchie** [comment]
That will allow you to install the app.

The fact remains that Apple dropped Instruments support in Xcode 8, and Appium uses Instruments to automate the device.

**Comment by romainlouvet** [comment]
yes https://forums.developer.apple.com/thread/48936 :(

**Comment by imurchie** [comment]
We are actively working on an XCUITest driver to integrate into Appium. It's not quite ready for prime time, though.

**Comment by shivakumar89** [comment]
Hello imurchie ,

I wanted to run automation -selenium-java scripts on new iOS 10.0 with Xcode 8  ..While I run IWDP I can see these error : 
ss.add_fd(3)
ss.recv fd=3 len=294
ss.recv fd=3 len=682
ss.add_server_fd(4)
Could not connect to lockdownd. Exiting.: Permission denied
ss.remove_server_fd(4)
Unable to attach 9285c264a00455134d31e698ed542463bdaff43c inspector

wanted to find out can we not use xcode8 beta with Appium 1.5.3 ?

thanks
Shiva

**Comment by VTINFOJames** [comment]
I've tried xCode 8 beta 3 with Appium 1.5.3 not knowing about the deprecation of UIAutomation in xCode. The obvious outcome is that Appium is unable to launch the application.

**Comment by imurchie** [comment]
Yes, unfortunately. 

There is good news, though. We are working hard on getting XCUITest supported. At the moment much of it works, but there are things to iron out. If you would like to try, you need to [use Appium from its source](https://github.com/appium/appium/blob/master/docs/en/contributing-to-appium/appium-from-source.md).

We would love to have more people trying it out, though it is definitely a work in progress.

**Comment by ericwengelking** [comment]
For the record Appium does work with Xcode 8 beta 3 if you are using appium-xcuitest-driver by building Appium from source

**Comment by meirtsvi** [comment]
From what I've read in the source code there is no support for running apps on real device, right?

**Comment by imurchie** [comment]
Running from source gives you all the capabilities that are the releases,
plus new development since the last release.

**Comment by meirtsvi** [comment]
Specifically for xcui driver, there is no support for running on real device yet. Here is a pr that was not yet integrated: https://github.com/appium/appium-xcuitest-driver/pull/69
I'm I missing something?

**Comment by imurchie** [comment]
Yes, for xxuitest real device support is waiting on that pr. It ought to be
merged early on Monday.

**Comment by JaniJegoroff** [comment]
@imurchie I'm trying to run Appium Ruby bindings tests against the latest code but I get stuck to `[XCUITest] Launching WebDriverAgent on the device` phase. Any tips how to proceed from there?

```
Janis-iMac:appium janijegoroff$ node .
[Appium] Welcome to Appium v1.5.3 (REV 068c4e5d0a7683956fb03c8ba81026106f016da7)
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"platformName":"iOS","platformVersion":"10.0","deviceName":"iPhone 6","app":"/Users/janijegoroff/projects/ruby_lib/ios_tests/UICatalog.app","automationName":"XCUITest"}}
[MJSONWP] Calling AppiumDriver.createSession() with args: [{"platformName":"iOS","pla...
[Appium] Creating new XCUITestDriver session
[Appium] Capabilities:
[Appium]   platformName: 'iOS'
[Appium]   platformVersion: '10.0'
[Appium]   deviceName: 'iPhone 6'
[Appium]   app: '/Users/janijegoroff/projects/ruby_lib/ios_tests/UICatalog.app'
[Appium]   automationName: 'XCUITest'
[BaseDriver] Session created with session id: 35062c79-a8bd-43dc-a149-699354e9bb3f
[iOSSim] Constructing iOS simulator for Xcode version 8.0 with udid '2CD5B95B-C8D4-42CB-BCC0-FF78C6C5F851'
[XCUITest] Determining device to run tests on: udid: '2CD5B95B-C8D4-42CB-BCC0-FF78C6C5F851', real device: false
[debug] [XCUITest] Checking whether app is actually present
[debug] [ios-app-utils] Getting bundle ID from app
[XCUITest] Setting up simulator
[XCUITest] Simulator with udid '2CD5B95B-C8D4-42CB-BCC0-FF78C6C5F851' not booted. Booting up now
[debug] [iOSSim] Killing all iOS Simulators
[iOSSim] Starting simulator with command: open /Applications/Xcode-beta.app/Contents/Developer/Applications/Simulator.app --args -CurrentDeviceUDID 2CD5B95B-C8D4-42CB-BCC0-FF78C6C5F851
[iOSSim] No boot indicator case for platform version '10.0'
[iOSSim] Tailing simulator logs until we encounter the string "no boot indicator string available"
[iOSSim] We will time out after 60000ms
[debug] [iOSSim] Simulator startup timed out. Continuing anyway.
[debug] [iOSSim] Waiting an extra 2000ms for the simulator to really finish booting
[debug] [iOSSim] Done waiting extra time for simulator
[iOSSim] Simulator booted in 62125ms
[XCUITest] Using default agent: /Users/janijegoroff/projects/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj
[XCUITest] Using default bootstrap: /Users/janijegoroff/projects/appium/node_modules/appium-xcuitest-driver/WebDriverAgent
[XCUITest] Launching WebDriverAgent on the device
```

**Comment by imurchie** [comment]
It seems the system log location for the newest beta of Xcode 8 moved, so Appium was not able to find the information it needed. Fixing.

There is also the issue that the boot indicator has changed, and I'm not able to find anything that seems reasonable to wait for. So, as is seen in the logs above, Appium waits 60s for the simulator to boot, even though it boots much more quickly than that usually.

If anyone wanted to dig into the logs a little bit, to see if there is anything that reliably comes up to indicate that the sim is actually booted, it would be much appreciated!

The indicators for previous versions are found here: https://github.com/appium/appium-ios-simulator/blob/master/lib/simulator-xcode-6.js#L189-L211

**Comment by skumar1122** [comment]
so now Will it work from NPM install or still need to use through source code ?

**Comment by imurchie** [comment]
It is still only available through the source, and it is still a work in progress.

**Comment by skumar1122** [comment]
awesome ! thank you

**Comment by rafalf** [comment]
I presume that python client will have to be rewritten for xcui? any predictions when will that be all integrated altogether in appium? i am about to start using this tool but uiautomation aint supported by apple any longer i recon i must wait for xcui

**Comment by imurchie** [comment]
Unfortunately, I am the main one working on implementing the XCUITest integration **and** the main one who takes care of the Python client, so it will be slow.

For the most part there will be no changes, other than sending direct UI Automation commands for  finding elements.

**Comment by Nesaganesh** [comment]
imurchie Do you have any idea when are you planning to complete this, are you anytime closer to the completion, sorry to trouble you, just wanted to know in curiosity

**Comment by lgjrb** [comment]
@imurchie,  Please update if the latest work on appium-xcuitest-driver supports testing on iOS 10 real devices. from "https://github.com/appium/appium-xcuitest-driver/pull/69", It only creates session and no other functionalities are supported.

**Comment by imurchie** [comment]
I don't understand your question.

**Comment by lgjrb** [comment]
Would like know, if the latest changes on appium supports testing all functionality on iOS 10 real devices.

**Comment by imurchie** [comment]
Not all functionality, no. We are working on finishing it. But most of it is there, minus touch actions.

**Comment by lgjrb** [comment]
great. Thanks much :-)

**Comment by SrinivasanTarget** [comment]
@triager please label: Enhancement

**Comment by rkang** [comment]
when is ETA for python support with XCUITest? I tried to run it with current python client and it crashed after the sim booted up with below message.

[XCUITest] Launching WebDriverAgent on the device
[XCUITest] Error: Command '/bin/bash Scripts/bootstrap.sh -d' exited with code 1
    at ChildProcess.<anonymous> (lib/teen_process.js:66:19)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:852:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
 { Error: Command '/bin/bash Scripts/bootstrap.sh -d' exited with code 1
    at ChildProcess.<anonymous> (lib/teen_process.js:66:19)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:852:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
  stdout: '\u001b[1mFetching dependencies\nPlease make sure that you have Carthage installed (https://github.com/Carthage/Carthage)\nNote: We are expecting that carthage installed in /usr/local/bin/\n',
  stderr: '',
  code: 1 }

**Comment by imurchie** [comment]
You need to install Carthage. [See the documentation](https://github.com/appium/appium-xcuitest-driver#external-dependencies). Or run `brew install carthage`

**Comment by JaniJegoroff** [comment]
@rkang I got same error at some point and installed Carthage manually.

**Comment by rkang** [comment]
thanks, brew install carthage fixed the issue, but now I am getting 'xcodebuild failed with code 65' error, any thought?

[XCUITest] xcodebuild exited with code '65' and signal 'null'
[XCUITest] Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:184:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
 Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:184:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
[XCUITest] Shutting down WebDriverAgent
[XCUITest] Simulator log exited with code 'null'

**Comment by imurchie** [comment]
There should be a log line like 

```
dbug XCUITest Log file for xcodebuild test: /Users/isaac/Library/Developer/Xcode/DerivedData/WebDriverAgent-eoyoecqmiqfeodgstkwbxkfyagll/Logs/Test/E395D6B2-0B14-4699-BA63-6BA5A3D8A385/Session-WebDriverAgentRunner-2016-08-23_105246-bt4zaE.log
```

Can you look in there for any information? You could post a link to a gist of it, if you want other eyes.

**Comment by rkang** [comment]
Thanks for the quick response, I found the below log from the location you provided.


10:41:59.183 xcodebuild[70775:1001102] Beginning test session WebDriverAgentRunner-4A83725F-004A-4DDB-87B4-77A0852E9E34 at 2016-08-23 10:41:59.181 with Xcode 8S201h on target <DVTiPhoneSimulator: 0x7f8a3b09f020> {
		SimDevice: SimDevice : iPad Air (B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898) : state={ Booted } deviceType={ SimDeviceType : com.apple.CoreSimulator.SimDeviceType.iPad-Air } runtime={ SimRuntime : 10.0 (14A5339a) - com.apple.CoreSimulator.SimRuntime.iOS-10-0 }
} (10.0 (14A5339a))
10:41:59.183 xcodebuild[70775:1001102] 	/Applications/Xcode-beta.App/Contents/Developer/usr/bin/xcodebuild
-project
/Applications/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj
-scheme
WebDriverAgentRunner
-sdk
iphonesimulator
-destination
platform=iOS Simulator,id=B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898
test
10:41:59.183 xcodebuild[70775:1001102] Launching with Xcode.IDEFoundation.Launcher.PosixSpawn
10:41:59.184 xcodebuild[70775:1001102] Initializing test infrastructure.
10:41:59.185 xcodebuild[70775:1001106] B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898: Registered for sim device notification, got token 4
10:41:59.185 xcodebuild[70775:1001106] B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898: Unregistering for sim device notification with token 4
10:41:59.185 xcodebuild[70775:1001106] Sim iPad Air is booted (state = 3)
10:41:59.185 xcodebuild[70775:1001106] Calling -[SimDevice getenv:error:] for TESTMANAGERD_SIM_SOCK
10:41:59.186 xcodebuild[70775:1001106] Returning /private/tmp/com.apple.launchd.gm9jksEKDn/com.apple.testmanagerd.unix-domain.socket for TESTMANAGERD_SIM_SOCK
10:41:59.186 xcodebuild[70775:1001106] Sim iPad Air has testmanagerd socket at /private/tmp/com.apple.launchd.gm9jksEKDn/com.apple.testmanagerd.unix-domain.socket
10:41:59.186 xcodebuild[70775:1001106] Connected socket 35 to testmanagerd for Sim iPad Air
10:41:59.186 xcodebuild[70775:1001106] Creating the test bundle connection.
10:41:59.187 xcodebuild[70775:1001106] Listening for proxy connection request from the test bundle (all platforms)
10:41:59.187 xcodebuild[70775:1001106] Resuming the connection.
10:41:59.188 xcodebuild[70775:1001106] Test connection requires daemon assistance.
10:41:59.194 xcodebuild[70775:1001102] Received Ready from iOSSimulator
10:41:59.194 xcodebuild[70775:1001102] Sending notification: NSConcreteNotification 0x7f8a3ae77480 {name = com.apple.iphonesimulator.startSession; userInfo = {
    deviceUDID = "B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898";
}}
10:41:59.306 xcodebuild[70775:1001102] Starting test session with ID <__NSConcreteUUID 0x7f8a3b1e01c0> 4A83725F-004A-4DDB-87B4-77A0852E9E34
10:41:59.306 xcodebuild[70775:1001102] Checking test manager availability..., will wait up to 120s
10:42:02.773 xcodebuild[70775:1001102] Test operation failure: Test operation was canceled.
10:42:02.773 xcodebuild[70775:1001102] _finishWithError:Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Test operation was canceled." UserInfo={NSLocalizedDescription=Test operation was canceled.} didCancel: 1

**Comment by imurchie** [comment]
Not particularly useful, then.

I've created an issue to work on getting the xcodebuild logs more easily. They are huge so printing them out obliterates anything else useful.

For now, there are a few steps to get them.

Go into your appium installation, on the command line. Then `cd node_modules/appium-xcuitest-driver`. Open, to edit, `lib/webdriveragent.js`. On around [line 134](https://github.com/appium/appium-xcuitest-driver/blob/master/lib/webdriveragent.js#L134) uncomment `// log.info(out);` (so it becomes `log.info(out);`). Save and exit.

Finally, run `gulp transpile` to get the working code.

Run appium again and you should get a lot more output, including, eventually, an error message.

My apologies for the difficulty in this. It is, as we say, a work in progress.

**Comment by rkang** [comment]
I did what you asked, but it doesn't seem much better than previous. 

[XCUITest] xcodebuild exited with code '65' and signal 'null'
[XCUITest] Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:184:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
 Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:184:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
[XCUITest] Shutting down WebDriverAgent
[XCUITest] Simulator log exited with code 'null'
[debug] [XCUITest] Resetting simulator
[debug] [XCUITest] Shutting down simulator
[debug] [iOSSim] Killing all iOS Simulators
[debug] [iOSLog] Stopping iOS log capture
[MJSONWP] Encountered internal error running command: Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:184:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
2016-08-23 11:24:46,663 - selenium.webdriver.remote.remote_connection - DEBUG - Finished Request
[HTTP] <-- POST /wd/hub/session 500 25435 ms - 167 
2016-08-23 11:24:46,663 - atom.atom_appium - ERROR - Message: An unknown server-side error occurred while processing the command. Original error: xcodebuild failed with code 65
Traceback (most recent call last):
  File "/school/product-tests/libraries/python/atom/atom/atom_appium/__init__.py", line 105, in start
    self._driver = WebDriverWrapper(command_executor=self.command_executor, desired_capabilities=capabilities)
  File "/school/product-tests/libraries/python/atom/atom/appium_utillities/driver_wrapper.py", line 24, in __init__
    super(WebDriverWrapper, self).__init__(command_executor, desired_capabilities)
  File "/usr/local/lib/python2.7/site-packages/appium/webdriver/webdriver.py", line 35, in __init__
    super(WebDriver, self).__init__(command_executor, desired_capabilities, browser_profile, proxy, keep_alive)
  File "/usr/local/lib/python2.7/site-packages/selenium-2.48.0-py2.7.egg/selenium/webdriver/remote/webdriver.py", line 87, in __init__
    self.start_session(desired_capabilities, browser_profile)
  File "/usr/local/lib/python2.7/site-packages/selenium-2.48.0-py2.7.egg/selenium/webdriver/remote/webdriver.py", line 141, in start_session
    'desiredCapabilities': desired_capabilities,
  File "/usr/local/lib/python2.7/site-packages/selenium-2.48.0-py2.7.egg/selenium/webdriver/remote/webdriver.py", line 201, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python2.7/site-packages/selenium-2.48.0-py2.7.egg/selenium/webdriver/remote/errorhandler.py", line 181, in check_response
    raise exception_class(message, screen, stacktrace)
WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: xcodebuild failed with code 65

2016-08-23 11:24:56,669 - selenium.webdriver.remote.remote_connection - DEBUG - POST http://127.0.0.1:4723/wd/hub/session {"desiredCapabilities": {"deviceName": "iPad Air", "language": "en", "simulator": true, "locale": "en_US", "app": "/school/product-tests/mobile-build/iPhoneSimulator/testable/Tableau.app", "udid": null, "platformVersion": "10.0", "platformName": "iOS", "appiumVersion": 1.4, "serverVersion": "9.0"}}
[HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"deviceName":"iPad Air","language":"en","simulator":true,"locale":"en_US","app":"/school/product-tests/mobile-build/iPhoneSimulator/testable/Tableau.app","udid":null,"platformVersion":"10.0","platformName":"iOS","appiumVersion":1.4,"serverVersion":"9.0"}}

**Comment by imurchie** [comment]
No idea what went wrong, then.

I have a fix for the logging going through the build. I'll note here when it gets published.

**Comment by imurchie** [comment]
@rkang If you update `appium-xcuitest-driver` to 2.0.10 you should get better logging of the xcode build failure.

**Comment by riadcse07** [comment]
i have an ios app which i want to test in i0S10 simulator . i installed xcode8 beta and using appium 1.4.13 version. but when i start to launch my app it shows can not find simualtor.

**Comment by imurchie** [comment]
You need Appium 1.6.0 beta 1 at least.

**Comment by riadcse07** [comment]
Thanx  imurchie.
 But currently latest version of appium is 1.5.3. So is there any other way to test my app in iOS 10 simulator or i should postpone until the beta 1.6.0 is released. i am currently running appium through a java application and testing my app . Waiting for your kind reply.

**Comment by JaniJegoroff** [comment]
1.6.0 beta is released 10 days ago.

**Comment by imurchie** [comment]
Use 'npm install -g appium@beta'

**Comment by riadcse07** [comment]
thanks @JaniJegoroff @imurchie . I have installed int Appium 1.6.0 beta 1 via the above command.

After that now Simulator loads and crashed with the below error message. 


[XCUITest] Waiting for WebDriverAgent to start on device
[debug] [XCUITest] Log file for xcodebuild test: /Users//Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/BB7FC2D6-E47C-455D-B799-27A4F00A2C6D/Session-WebDriverAgentRunner-2016-08-29_171054-IEPucH.log
[XCUITest] 2016-08-29 17:11:08.109 xcodebuild[1110:20771] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Test operation was canceled. If you believe this error represents a bug, please attach the log file at /Users//Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/BB7FC2D6-E47C-455D-B799-27A4F00A2C6D/Session-WebDriverAgentRunner-2016-08-29_171054-IEPucH.log" UserInfo={NSLocalizedDescription=Test operation was canceled. If you believe this error represents a bug, please attach the log file at /Users/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/BB7FC2D6-E47C-455D-B799-27A4F00A2C6D/Session-WebDriverAgentRunner-2016-08-29_171054-IEPucH.log}

[XCUITest] 2016-08-29 17:11:08.174 xcodebuild[1110:22251] Connection peer refused channel request for "dtxproxy:XCTestManager_IDEInterface:XCTestManager_DaemonConnectionInterface"; channel canceled <DTXChannel: 0x7ffcc2e779d0>

[XCUITest] 
Testing failed:

[XCUITest] 	./Scripts/generate_modules.sh: line 15: Modules/module.modulemap: Permission denied
** TEST FAILED **


The following build commands failed:
	PhaseScriptExecution Generate\ modules /Users//Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Build/Intermediates/WebDriverAgent.build/Debug-iphonesimulator/WebDriverAgentLib.build/Script-AD35D0281CF1B37A00870A75.sh
(1 failure)

[XCUITest] xcodebuild exited with code '65' and signal 'null'
[debug] [XCUITest] xcodebuild failure warrants retry. Retrying...
[XCUITest] Shutting down WebDriverAgent
[XCUITest] Simulator log exited with code 'null'
[XCUITest] Launching WebDriverAgent on the device
[debug] [XCUITest] Beginning test with command 'xcodebuild -project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=B04752D8-0CFD-4CA9-904B-BD0EA244DED3 test' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'
[XCUITest] Waiting for WebDriverAgent to start on device
[debug] [XCUITest] Log file for xcodebuild test: /Users//Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/C13DE6BD-B285-4B65-A402-86D85F5E7F26/Session-WebDriverAgentRunner-2016-08-29_171211-GEzj8O.log
[XCUITest] 2016-08-29 17:12:18.858 xcodebuild[1158:22357] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Test operation was canceled. If you believe this error represents a bug, please attach the log file at /Users//Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/C13DE6BD-B285-4B65-A402-86D85F5E7F26/Session-WebDriverAgentRunner-2016-08-29_171211-GEzj8O.log" UserInfo={NSLocalizedDescription=Test operation was canceled. If you believe this error represents a bug, please attach the log file at /Users//Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/C13DE6BD-B285-4B65-A402-86D85F5E7F26/Session-WebDriverAgentRunner-2016-08-29_171211-GEzj8O.log}

[XCUITest] 2016-08-29 17:12:18.859 xcodebuild[1158:22854] Connection peer refused channel request for "dtxproxy:XCTestManager_IDEInterface:XCTestManager_DaemonConnectionInterface"; channel canceled <DTXChannel: 0x7fb56ca89f80>

[XCUITest] 
Testing failed:

[XCUITest] 	./Scripts/generate_modules.sh: line 15: Modules/module.modulemap: Permission denied
** TEST FAILED **


The following build commands failed:
	PhaseScriptExecution Generate\ modules /Users//Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Build/Intermediates/WebDriverAgent.build/Debug-iphonesimulator/WebDriverAgentLib.build/Script-AD35D0281CF1B37A00870A75.sh
(1 failure)

[XCUITest] xcodebuild exited with code '65' and signal 'null'
[XCUITest] Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:214:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
 Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:214:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
[XCUITest] Shutting down WebDriverAgent
[XCUITest] Simulator log exited with code 'null'
[debug] [XCUITest] Resetting simulator
[debug] [XCUITest] Shutting down simulator
[debug] [iOSSim] Killing all iOS Simulators
[debug] [iOSLog] Stopping iOS log capture
[MJSONWP] Encountered internal error running command: Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:214:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
[HTTP] <-- POST /wd/hub/session 500 345549 ms - 167

**Comment by ssynix** [comment]
[Set Xcode beta as default](http://stackoverflow.com/questions/25910796/how-to-set-xcode-version-as-default-in-my-mac). That worked for me.

**Comment by briananderson1222** [comment]
@ssynix That helped :) now i'm at `[XCUITest] xcodebuild exited with code '65' and signal 'null'
` which I think is related to the signing with the same team issue

**Comment by ssynix** [comment]
For that I had to open the WDA Xcode project and [set the team settings manually](https://github.com/facebook/WebDriverAgent/wiki/Common-Issues).

**Comment by briananderson1222** [comment]
Once again that worked ;) now the app is installing but appium seems to freeze at
```
[XCUITest] Waiting for WebDriverAgent to start on device
```
On the device the WebDriver app starts then exits immediately.. any experience with this? I am so close! :D .. Manually opening the app freezes with a black screen and what looks like a battery symbol a third of the way down the screen..

>

**Comment by yaphatak** [comment]
I think deviceconsole is missing. Can you also do this as per doc
https://github.com/appium/appium-xcuitest-driver#usage

**Comment by yaphatak** [comment]
idevicesyslog doesn't work with ios 10 (at least until last time I tried)

**Comment by briananderson1222** [comment]
Okay, I tried with `deviceconsole`

The only difference is now I get this debug output:

```
[debug] [XCUITest] Waiting for WebDriverAgent server to finish loading...
[debug] [WebDriverAgent] Device: Sep  2 11:54:00 XXX-My-iPhone XCTRunner[448] <Notice>: Running tests...
[debug] [XCUITest] Waiting for WebDriverAgent server to finish loading...
[debug] [XCUITest] Waiting for WebDriverAgent server to finish loading...
[debug] [WebDriverAgent] Device: Sep  2 11:54:01 XXX-My-iPhone XCTRunner[448] <Notice>: Continuing to run tests in the background with task ID 1
[debug] [XCUITest] Waiting for WebDriverAgent server to finish loading...
```

It continuously repeats that last line over and over.. the WDA app still seems to exit right away.

I appreciate the help

**Comment by yaphatak** [comment]
can u run this from your WebDriverAgent folder , xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination 'id=<udid>' test.

**Comment by briananderson1222** [comment]
@yaphatak I think there is a typo in the `'id=' test.` part of your string.. Should it be just `id=test`?

**Comment by ssynix** [comment]
WDA doesn't really have a user interface (hence the black screen), so it just runs in the background. I think you're stuck because the regex that parses the logs is looking for a 2-digit number for day of the month, but WDA pads the day with a space instead of a zero (so looking for `Sept 02` and seeing `Sept  2`). https://github.com/appium/appium-xcuitest-driver/pull/118

**Comment by yaphatak** [comment]
xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination 'id=<udid>' test

**Comment by briananderson1222** [comment]
ha.. okay. And this would prevent the tests from even starting? from the front-end side of this I am freezing at `> Initializing Appium session: { host: 'localhost', port: 4723 }` while the server keeps printing out the 'waiting' message from above.

**Comment by briananderson1222** [comment]
@yaphatak I get this output:

```
2016-09-02 12:17:35.401429 XCTRunner[504:115513] Running tests...
2016-09-02 12:17:37.007800 XCTRunner[504:115513] Continuing to run tests in the background with task ID 1

Restarting after unexpected exit or crash in UITestingUITests/testRunner; summary will include totals from previous launches.

Test Suite 'Selected tests' started at 2016-09-02 12:17:37.133
Test Suite 'WebDriverAgentRunner.xctest' started at 2016-09-02 12:17:37.136
Test Suite 'WebDriverAgentRunner.xctest' failed at 2016-09-02 12:17:37.137.
       	 Executed 1 test, with 1 failure (0 unexpected) in 0.000 (0.001) seconds
Test Suite 'WebDriverAgentLib.framework' started at 2016-09-02 12:17:37.139
Test Suite 'WebDriverAgentLib.framework' passed at 2016-09-02 12:17:37.140.
       	 Executed 0 tests, with 0 failures (0 unexpected) in 0.000 (0.001) seconds
Test Suite 'Selected tests' failed at 2016-09-02 12:17:37.144.
       	 Executed 1 test, with 1 failure (0 unexpected) in 0.000 (0.011) seconds
Failing tests:
       	-[UITestingUITests testRunner]
** TEST FAILED **
```

**Comment by ssynix** [comment]
For me the `startDetector` code wouldn't detect the server address because it tried looking for `Built at <date>` first. I'm not sure why it doesn't just look for `ServerURLHere` but I trust that there was a good reason for it 😅

**Comment by yaphatak** [comment]
@briananderson1222 Look like WDA itself is not working. Are you using xcode 8 beta 6?

**Comment by briananderson1222** [comment]
@yaphatak I am on Xcode 8 Beta 6 

@ssynix I will take a look and see if my issue is similar

Thanks for the help guys

**Comment by briananderson1222** [comment]
Just to try and get around this issue @ssynix  for now I updated my system clock and phone clock to the Sep 10th.

Now when running @yaphatak's suggested `xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination 'id=UDID' test` I get the following output that seems to freeze:

```
2016-09-10 11:01:30.667285 XCTRunner[239:4967] Running tests...
2016-09-10 11:01:32.288360 XCTRunner[239:4967] Continuing to run tests in the background with task ID 1
Test Suite 'All tests' started at 2016-09-10 11:01:32.401
Test Suite 'WebDriverAgentRunner.xctest' started at 2016-09-10 11:01:32.403
Test Suite 'UITestingUITests' started at 2016-09-10 11:01:32.406
Test Case '-[UITestingUITests testRunner]' started.
    t =     0.00s     Start Test at 2016-09-10 11:01:32.411
    t =     0.01s     Set Up
2016-09-10 11:01:32.419930 XCTRunner[239:4967] Built at Sep 10 2016 10:01:11
2016-09-10 11:01:32.457309 XCTRunner[239:4967] ServerURLHere->http://MY-IP-ADDR-HERE:8100<-ServerURLHere
2016-09-10 11:01:32.458809 XCTRunner[239:4967] Listening on USB
```

**Comment by rkang** [comment]
Hi @imurchie,

I made some progress and I am able to install the WebDriverAgent, my testing app and open, but when I try to open the connection with the appium, it crash. Any thoughts?


[iOSLog] [IOS_SYSLOG_ROW] Sep  7 15:54:42 mobilelab-mac8 assertiond[28545]: notify_suspend_pid() failed with error 7
[XCUITest] Error: Could not proxy command to remote server. Original error: Error: socket hang up
    at JWProxy.proxy$ (lib/jsonwp-proxy/proxy.js:115:13)
    at tryCatch (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
 Error: Could not proxy command to remote server. Original error: Error: socket hang up
    at JWProxy.proxy$ (lib/jsonwp-proxy/proxy.js:115:13)
    at tryCatch (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
[XCUITest] Shutting down WebDriverAgent
[XCUITest] Simulator log exited with code 'null'
[iOSLog] [IOS_SYSLOG_ROW] Sep  7 15:54:42 mobilelab-mac8 installd[28536]: 0x70000019a000 -[MIInstallableBundle _refreshUUIDForContainer:withError:]: Data container for com.apple.test.WebDriverAgentRunner-Runner is now at /Users/tabauto/Library/Developer/CoreSimulator/Devices/B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898/data/Containers/Data/Application/774DBF68-B992-4495-BC85-39AF93D9336F
[iOSLog] [IOS_SYSLOG_ROW] Sep  7 15:54:42 mobilelab-mac8 installd[28536]: 0x70000019a000 -[MIInstallableBundle _refreshUUIDForContainer:withError:]: Data container for com.apple.test.WebDriverAgentRunner-Runner is now at /Users/tabauto/Library/Developer/CoreSimulator/Devices/B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898/data/Containers/Data/Application/774DBF68-B992-4495-BC85-39AF93D9336F
[iOSLog] [IOS_SYSLOG_ROW] Sep  7 15:54:42 mobilelab-mac8 installd[28536]: 0x70000019a000 -[MIInstallableBundle _refreshUUIDForContainer:withError:]: Data container for com.apple.test.WebDriverAgentRunner-Runner is now at /Users/tabauto/Library/Developer/CoreSimulator/Devices/B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898/data/Containers/Data/Application/774DBF68-B992-4495-BC85-39AF93D9336F
[XCUITest] xcodebuild exited with code 'null' and signal 'SIGTERM'
[debug] [XCUITest] Resetting simulator
[debug] [iOSLog] Stopping iOS log capture
[iOSLog] [IOS_SYSLOG_ROW] Sep  7 15:54:42 mobilelab-mac8 installd[28536]: 0x70000019a000 -[MIContainer makeContainerLiveReplacingContainer:reason:withError:]: Made container live for com.apple.test.WebDriverAgentRunner-Runner at /Users/tabauto/Library/Developer/CoreSimulator/Devices/B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898/data/Containers/Bundle/Application/73F8D2E9-BBF3-45F9-871E-F514A5F9DB2C
[iOSLog] [IOS_SYSLOG_ROW] Sep  7 15:54:42 mobilelab-mac8 installd[28536]: 0x70000019a000 -[MIContainer makeContainerLiveReplacingContainer:reason:withError:]: Made container live for com.apple.test.WebDriverAgentRunner-Runner at /Users/tabauto/Library/Developer/CoreSimulator/Devices/B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898/data/Containers/Bundle/Application/73F8D2E9-BBF3-45F9-871E-F514A5F9DB2C
[iOSLog] [IOS_SYSLOG_ROW] Sep  7 15:54:42 mobilelab-mac8 installd[28536]: 0x70000019a000 -[MIContainer makeContainerLiveReplacingContainer:reason:withError:]: Made container live for com.apple.test.WebDriverAgentRunner-Runner at /Users/tabauto/Library/Developer/CoreSimulator/Devices/B9D0F158-5BC0-4F4C-A9B0-92A65CD4E898/data/Containers/Bundle/Application/73F8D2E9-BBF3-45F9-871E-F514A5F9DB2C
[MJSONWP] Encountered internal error running command: Error: Could not proxy command to remote server. Original error: Error: socket hang up
    at JWProxy.proxy$ (lib/jsonwp-proxy/proxy.js:115:13)
    at tryCatch (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/appium/node_modules/appium-xcuitest-driver/node_modules/babel-runtime/regenerator/runtime.js:136:37)
[HTTP] <-- POST /wd/hub/session 500 15679 ms - 216 
2016-09-07 15:54:42,433 - selenium.webdriver.remote.remote_connection - DEBUG - Finished Request
2016-09-07 15:54:42,433 - atom.atom_appium - ERROR - Message: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to remote server. Original error: Error: socket hang up

**Comment by pavanbachu0604** [comment]
I am getting the " _Original error: simctl error running 'create': Invalid runtime: undefined (WARNING: The server did not provide any stacktrace information). Command duration or timeout: 366 milliseconds"._  I am trying to run the appium 1.6 on iOS 10 real device. Has the appium 1.6 beta working on real devices ?!

**Comment by yaphatak** [comment]
@rkang : Can you see if you have `iproxy` install on your machine? just type `iproxy` on terminal and see if it gives any output

**Comment by yaphatak** [comment]
@pavanbachu0604 : Can you provide what capabilities you passed?

**Comment by skumar1122** [comment]
Heya,
Using:      Xcode 8 
                 Appium 1.1.6.0-beta1
                 iPhone6Plus 10.0 
              
I have appium 1.6.0-beta1 with iOS 10.0 Iphone6Plus device and I get following error when i start my IWDP ....

---appium -U 9285c26045513fsdfdsf98ed542463bdaff43c -p4723 --full-reset  --webkit-debug-proxy-port 27790
---ios_webkit_debug_proxy -c 9285c26045513fsdfdsf98ed542463bdaff43c:27790 -d

ERROR -
ss.add_fd(3)
ss.recv fd=3 len=977
ss.add_server_fd(4)
Could not connect to lockdownd. Exiting.: Broken pipe
ss.remove_server_fd(4)
Unable to attach 9285c26045513fsdfdsf98ed542463bdaff43c inspector
^Css.remove_fd(3)

Any one seen this before ?

**Comment by pavanbachu0604** [comment]
@yaphatak  
I am trying to run on iPad Mini 4 and iOS 10 with appium 1.6 
capabilities.setCapability("platformVersion", "10.0");
capabilities.setCapability("deviceName", "iPad");
 capabilities.setCapability("platform", "iOS");
 capabilities.setCapability("PLATFORM_VERSION", "10.0");
 capabilities.setCapability("automationName", "XCUITest");
capabilities.setCapability("app", "com.vzw.pos.native");
driver.set(new IOSDriver(new URL("http://localhost:4723/wd/hub"),capabilities))

**Comment by pavanbachu0604** [comment]
@skumar1122 
1. Please make sure that you have the "Enable UI Automation" active under developer options.
2. In safari > advanced options> make sure web inspector is active 
If none of two options work please try the steps from the given link : 
https://gist.github.com/pavanbachu0604/d505fa1978e45d95ab4ba44c8fb17d44.

**Comment by skumar1122** [comment]
@pavanbachu0604 
Yes Enabled UI Automation and WebInspector is Active !!!
I have done that as a basic , It worked until 9.3.5 with Xcode 7 on any real devices.
I do have Appium - Mac Properties :
Xcode: 8
OSX: 10.11.6
iOS 10.0
Installed though homebrew,
Appium 1.1.6.0-beta1
libimobiledevice-1.2.0
ideviceinstaller-1.1.0_2
ios-webkit-debug-proxy-1.7
I have created provisional profile for my devices too...did i missed any thing in xcode8 ?

**Comment by pavanbachu0604** [comment]
@skumar1122 : Please uninstall  the ideviceinstaller and iso-webkit-debug-proxy from mac then reinstall them from the commands mentioned by me earlier. In Xcode 8; the support for UI Automation is deprecated so please use this capability : capabilities.setCapability("automationName", "XCUITest"); and Appium 1.6 right now supports simulator only.

**Comment by briananderson1222** [comment]
@skumar1122 Here is what I have documented so far.. I suppose some of this could be out of date already.. I am still stuck at the point I described above, but maybe this will help you get pass some of your issues

**Steps to run**:

- `[sudo ]npm install -g appium@beta`
- `sudo chmod 777 /var/db/lockdown` (from https://github.com/libimobiledevice/libimobiledevice/issues/356)
- `brew uninstall ios-webkit-debug-proxy` (follow the instructions here https://github.com/google/ios-webkit-debug-proxy for building from source)
- `brew uninstall libimobiledevice && brew install --HEAD libimobiledevice`   
- `brew install usbmuxd`
- `brew install carthage` (WDA project uses this)
- https://github.com/appium/appium-xcuitest-driver#usage -- ensure deviceconsole is pulled down, compiled, and `realDeviceLogger` referenced in your appium capabilities
- https://github.com/facebook/WebDriverAgent/wiki/Common-Issues -- ensure the provisioning profile or team is set on the WDA project to match your project

**Make sure you can run (without errors)**:

- `idevicepair pair` (Hit 'Trust' on device)
- `idevicepair validate` 
- `ios_webkit_debug_proxy --debug` (just make sure this runs without error then you can ctrl+c)

**Modify `appium` capabilities to include**:

- `platformVersion: '10'`
- `automationName: 'XCUITest'`
- `realDeviceLogger: <path/to/deviceconsole>`  # requires you to run `make` after pulling down deviceconsole repo

**Comment by pavanbachu0604** [comment]
I understand that you have the updated ideviceinstaller and iOS-webkit-debug proxy, but I strongly recommend you uninstalling/deleting both of them and reinstalling from the source(as mentioned in link provided by me earlier). The "chmod" does not work when you are working on iOS 10.

**Comment by yaphatak** [comment]
@pavanbachu0604 : can you remove second `PLATFORM_VERSION`. error message that you are getting => `platformVersion` is missing

**Comment by pavanbachu0604** [comment]
@yaphatak : I am still getting the same error

**Comment by JamesRootNineAgile** [comment]
The key thing that fixes the problem is
**sudo chmod 777 /var/db/lockdown**
This won't survive a reboot and the problem returns, related to https://github.com/libimobiledevice/libimobiledevice/issues/356

**Comment by briananderson1222** [comment]
@pavanbachu0604 the steps I provided are the steps I had to take to get to the current state mentioned in https://github.com/appium/appium/issues/6597#issuecomment-245317706 . Like I said these could be out of date but helped me get passed problems that arose while attempting to run our tests on a real device.

**Comment by skumar1122** [comment]
@pavanbachu0604 @JamesRootNineAgile 
Thank you guys for help,

I'm trying to do some real device Safari testing, it works on simulator but no on a real device whit iOS 10. I get this error message:

> WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: real device safari support not yet implemented. Contact an appium dev

So besides following all the steps from @briananderson1222, I cloned the WebDriverAgent source repo and replace the one provided in the npm appium dependency. 

Anyone have tried something for Safari?

**Comment by yaphatak** [comment]
@skumar1122 It wont work now there is change that need to be done on appium side.  And besides that you need to install ios-debug-proxy from head brew install --HEAD

**Comment by qakapil** [comment]
Hi, 
I followed the steps mentioned by @skumar1122 above and had some sort of success in setting up appium-xcuitest-driver.  Here is my environment info - 
appium-xcuitest-driver@2.0.12.
Xcode 7.3.2
iPhone iOS 9.1 (Will try on iOS10 in a few days. 
In WebDriverAgent.xcodeproj/project.pbxproj I had to change values of IPHONEOS_DEPLOYMENT_TARGET from 9.3 to 9.1

I also set following two values in desiredCapablities - 
'automationName': "xcuitest",
'realDeviceLogger':'/usr/local/bin/idevicesyslog'

After this I tried running my old python appium script. It did install the App on the device, launched the app, 
self.driver.find_element(*self.username_locator).clear() - worked fine
self.driver.find_element(*self.sign_in_locator).click() - worked fine
However, 
self.driver.find_element(*self.username_locator).send_keys('blablabla) is failing.

Is it because send_keys() method is using uiautomation ? 
Or because I am  using iOS9.1 ? 
Or maybe because I am on Xcode 7.3.1 ? 

Regards,
Kapil.

**Comment by acdurai** [comment]
@imurchie 

am getting the below mentioned error while running the test in ios 10

xcode 8
appium 1.6.0 beta
ios 10

at GeneratorFunctionPrototype.invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
[XCUITest] Shutting down WebDriverAgent
[debug] [XCUITest] Resetting simulator
[debug] [iOSLog] Stopping iOS log capture
**[MJSONWP] Encountered internal error running command: Error: Unable to find real device logging program 'deviceconsole'**
    at WebDriverAgent.createRealDeviceLogsSubProcess$ (lib/webdriveragent.js:188:13)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
[HTTP] <-- POST /wd/hub/session 500 5043 ms - 195

**Comment by imurchie** [comment]
https://github.com/appium/appium-xcuitest-driver#usage

**Comment by babusurendra** [comment]
Hi @imurchie , what is the expected date to release appium 1.6 goldmaster?

**Comment by acdurai** [comment]
@imurchie 

Thanks for sharing the link, that issue fixed am getting below issue

	Signing for "WebDriverAgentRunner" requires a development team. Select a development team in the project editor.
	Code signing is required for product type 'UI Testing Bundle' in SDK 'iOS 10.0'
** TEST FAILED **


The following build commands failed:
	Check dependencies
(1 failure)

[XCUITest] xcodebuild exited with code '65' and signal 'null'
[XCUITest] Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:236:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
 Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:236:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
[XCUITest] Shutting down WebDriverAgent
[XCUITest] System log exited with code '0'
[debug] [XCUITest] Resetting simulator
[debug] [iOSLog] Stopping iOS log capture
[MJSONWP] Encountered internal error running command: Error: xcodebuild failed with code 65
    at SubProcess.<anonymous> (lib/webdriveragent.js:236:25)
    at emitTwo (events.js:106:13)
    at SubProcess.emit (events.js:191:7)
    at ChildProcess.<anonymous> (lib/teen_process.js:191:14)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
[HTTP] <-- POST /wd/hub/session 500 59341 ms - 167

**Comment by imurchie** [comment]
Yes. Things need to be signed correctly.

https://github.com/appium/appium-xcuitest-driver/issues/106

**Comment by iamola** [comment]
@imurchie  Any update or workaround on this issue? My team and I saw this way too late.

**Comment by imurchie** [comment]
I'm sorry. This is a 80+ comment thread with no cohesion. I have no idea
what "this issue" is anymore.

**Comment by imurchie** [comment]
I'm sorry. This is a 80+ comment thread with no cohesion. I have no idea
what "this issue" is anymore.

**Comment by iamola** [comment]
@imurchie  "The fact remains that Apple dropped Instruments support in Xcode 8, and Appium uses Instruments to automate the device."

**Comment by imurchie** [comment]
Yes. We are hard at work to provide the same level of support with the new
tools.

**Comment by JaniJegoroff** [comment]
@imurchie I'm trying out the latest beta build with Ruby bindings. So far I have come across with one major issue:
```
timed out after 30 seconds (underlying webdriver instance does not support javascript) (Selenium::WebDriver::Error::TimeOutError)
```

There are couple of Ruby binding methods that ends up calling `execute_script` method:
https://github.com/appium/ruby_lib/blob/master/lib/appium_lib/ios/helper.rb#L554

I guess this is a same issue as https://github.com/appium/appium/issues/5623
I would like to tackle this issue since it fails so many tests.

**Comment by shay777** [comment]
I find this thread very useful but I still have one issue that was not resolved.
I use the beta version of appium 6, xcode 8 and ios 10. I cannot install my app on the device.
I had some of the errors that were mention here and I followed the solutions so there is a progress but now I have the error of:
Encountered internal error running command: Error: xcodebuild failed with code 65
Any idea how to solve this error?

**Comment by qakapil** [comment]
Hi @imurchie,
I build appium from source (master branch) and now I am able to launch WebdriverAgent and run my tests on iOS10 device. I am running this with Xcode 8. There is one minor issue which I am facing though. Even though WebdriverAgent gets installed on the device and my tests starts executing, on the console of appium logs, I continuously see this log message
[debug] [XCUITest] Waiting for WebDriverAgent server to finish loading...

I only see this log message with iOS10 device and not with iOS9. 
I guess some logic needs to be changed in lib/webdriveragent.js

**Comment by shay777** [comment]
is there any solution for this error:
Signing for "WebDriverAgentRunner" requires a development team. Select a development team in the project editor.
	Code signing is required for product type 'UI Testing Bundle' in SDK 'iOS 10.0'

**Comment by yaphatak** [comment]
Can you make sure your xcode project editor for `WDA` looks like this 
https://github.com/appium/appium-xcuitest-driver/issues/106#issuecomment-243001378

**Comment by shay777** [comment]
thanks for your reply.
it's still not working.
I have the same error message

**Comment by chaoleona** [comment]
I has updated appium to 1.6.0 beta1, xcode to 10, and run my test app in my iPhone SE, but the tap or click take much more time than before. A click may take 40s.   Can I improve this problem?

**Comment by jlipps** [comment]
Since this thread has devolved into reporting of many different issues, I am closing it. If your particular issue is not yet resolved, please open a new issue and ensure to use the template to give us all the information which is specific to _your_ issue. That will help us triage and fix as many things as we can with the new XCUITest support.

**Comment by Hellengo** [comment]
$sudo npm install --global --unsafe-perm ios-deploy
...
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dsymutil /Users/Hellengo/.npm-packages/lib/node_modules/.staging/ios-deploy-419d7fc2/build/Release/ios-deploy -o /Users/Hellengo/.npm-packages/lib/node_modules/.staging/ios-deploy-419d7fc2/build/Release/ios-deploy.dSYM

** BUILD SUCCEEDED **

/Users/Hellengo/.npm-packages/bin/ios-deploy -> /Users/Hellengo/.npm-packages/lib/node_modules/ios-deploy/build/Release/ios-deploy
/Users/Hellengo/.npm-packages/lib
└── ios-deploy@1.9.0

Run test:
selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Could not initialize ios-deploy make sure it is installed and works on your system
Log:
[MJSONWP] Calling AppiumDriver.createSession() with args: [{"deviceName":"iPhone 6","...
[Appium] Creating new XCUITestDriver session
[Appium] Capabilities:
[Appium]   deviceName: 'iPhone 6'
[Appium]   udid: '11ebe7ca5241c034fbe1845982da4fc79c68eb71'
[Appium]   autoAcceptAlerts: true
[Appium]   app: '/Users/Hellengo/Desktop/Test/vodka.ipa'
[Appium]   platformVersion: '9.3'
[Appium]   platformName: 'iOS'
[Appium]   automationName: 'XCUITest'
[BaseDriver] Session created with session id: 341bc5e6-ac96-41de-98b4-7bf140991cd4
[debug] [XCUITest] Xcode version set to '8.0'
[debug] [XCUITest] iOS SDK Version set to '10.0'
[debug] [XCUITest] Available devices: 80d160000eeeb4b4f1d6523a4d4ab762ae3197e6, 11ebe7ca5241c034fbe1845982da4fc79c68eb71
[debug] [XCUITest] Creating iDevice object with udid 11ebe7ca5241c034fbe1845982da4fc79c68eb71
[XCUITest] Could not initialize ios-deploy make sure it is installed and works on your system
[XCUITest] Error: Could not initialize ios-deploy make sure it is installed and works on your system
    at XCUITestDriver.getIDeviceObj$ (lib/driver.js:618:13)
    at tryCatch (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
 Error: Could not initialize ios-deploy make sure it is installed and works on your system
    at XCUITestDriver.getIDeviceObj$ (lib/driver.js:618:13)
    at tryCatch (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
[MJSONWP] Encountered internal error running command: Error: Could not initialize ios-deploy make sure it is installed and works on your system
    at XCUITestDriver.getIDeviceObj$ (lib/driver.js:618:13)
    at tryCatch (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Users/Hellengo/.npm-packages/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
